src/llvm_backend.cpp

```diff
@@ -1825,25 +1825,45 @@ gb_internal lbProcedure *lb_create_main_procedure(lbModule *m, lbProcedure *star
 		TEMPORARY_ALLOCATOR_GUARD();
 		auto args = array_make<lbValue>(temporary_allocator(), 1);
 		args[0] = lb_addr_load(p, all_tests_slice);
-		lb_emit_call(p, runner, args);
+		lbValue result = lb_emit_call(p, runner, args);
+
+		lbBlock *block_success = lb_create_block(p, "success");
+		lbBlock *block_failure = lb_create_block(p, "failure");
+
+		lbValue result_success = lb_emit_comp(p, Token_CmpEq, result, lb_const_bool(m, t_bool, true));
+		lb_emit_if(p, result_success, block_success, block_failure);
+
+		lbValue exit_runner = lb_find_package_value(m, str_lit("os"), str_lit("exit"));
+
+		lb_start_block(p, block_success);
+		{
+			auto exit_args = array_make<lbValue>(temporary_allocator(), 1);
+			exit_args[0] = lb_const_int(m, t_int, 0);
+			lb_emit_call(p, exit_runner, exit_args);
+		}
+
+		lb_start_block(p, block_failure);
+		{
+			auto exit_args = array_make<lbValue>(temporary_allocator(), 1);
+			exit_args[0] = lb_const_int(m, t_int, 1);
+			lb_emit_call(p, exit_runner, exit_args);
+		}
```

Comment:
Why not use a `lb_emit_select`?

```cpp
lbValue result = lb_emit_call(p, runner, args);
lbValue exit_runner = lb_find_package_value(m, str_lit("os"), str_lit("exit"));

auto exit_args = array_make<lbValue>(temporary_allocator(), 1);
exit_args[0] = lb_emit_select(p, result, lb_const_int(m, t_int, 1), lb_const_int(m, t_int, 0));
lb_emit_call(p, exit_runner, exit_args);
```

Reply:
I did not see that one, that is indeed clearer, updated it now